### ISSUE TITLE
Exploration of YOLO v3 For Object Detection

### ISSUE THREAD

**Issue by Nike1421** [UCI-SAP-Capstone-2024/Proof-of-Performance-Validation#19]
We'll be using YOLOv3, an object detection model, to detect beverage cans and bottles from promotion images.

**Comment by Nike1421** [comment]
During Sprint #1 (Apr 9 - Apr 22), I designed a object detection model using the YOLOv5 Object Detection Model aimed at detecting Beverage Cans from photos of retail promotions.

I used the model developed by vencerlanz09 on kaggle (Link: https://www.kaggle.com/code/vencerlanz09/cans-image-classification-using-yolov5) and trained the model on a Cans Detection Dataset (Link: https://www.kaggle.com/datasets/vencerlanz09/tin-and-steel-cans-synthetic-image-dataset). 

Due to limited processing power on my machine, I was left with no option than to use the smallest YOLO model (YOLO v5 small) for training. This comes with a penalty of lower overall accuracy for object detection.

After training the model for 10 epochs on the dataset, I used the test image from the dataset for checking the accuracy of the model. The results of that test are as follows:

![Image](https://github.com/UCI-SAP-Capstone-2024/Proof-of-Performance-Validation/assets/67200520/ca7f51dc-d3fe-4f62-b11f-b0fda9e9c31f)

After using the test image, I used a test image of my own which is closer to our application and tasks. The results are as follows:

![Image](https://github.com/UCI-SAP-Capstone-2024/Proof-of-Performance-Validation/assets/67200520/f6c9ec49-a38b-47b5-a07f-9f2b46fe9dbf)

This is a best case scenario in our task. The cans are of different types and are clearly separated. Thus, our model has little problem detecting the cans. However, using an image which is much closer to our use-case leads to the following result:

![Image](https://github.com/UCI-SAP-Capstone-2024/Proof-of-Performance-Validation/assets/67200520/1c121723-0ab1-42e8-9371-cac645b733d6)

As you can see clearly, not all the cans are detected, and for the ones which have been detected, the confidence score is sub-optimal. One of the reasons for this is that we used the weakest YOLOv5 Model for training. If we had used a more robust model, we would have got even more accurate results for the images.

Takeaways:

1. The YOLO Object Detection Model is a good object detection model for our purpose. 
2. If we had more processing power, we could have gotten more accurate results.
3. Considering the remaining timeline as well as the progress of other team implementing object detection models, it would take a long time to fine-tune the YOLO model and achieve a good enough accuracy for our needs.

Conclusion:
We will halt the development of the YOLOv5 Model for can detection and go forward with the FaceBook Detectron Model.
